src/data.ts

```diff
@@ -12,32 +21,61 @@ export default {
   ExternalERC721: {
     ledger: env.VITE_SC_EXTERNAL_ERC721_LEDGER_CONTRACT_ADDRESS,
     title: 'Mainnet NFT derivatives',
+    proofTitle: 'Mainnet NFTs',
     ownerTitle: 'This wallet owns a {derivative}',
     ownerContent:
       'This is a zkNFT derivative. It means this person has been verified to own at least one ‘{original}‘ Mainnet NFT.',
     proofIcon: Erc721Badge,
-
     badgeType: BadgeSourceType.ERC721,
     network: Network.Mainnet,
+    mint: async (provider: Web3Provider, proof: Proof) => {
+      const signature = await requestContractMetadata(
+        Network.Mainnet,
+        proof.origin
+      )
+      return createExternalERC721Badge(
+        provider,
+        proof,
+        signature.message,
+        signature.signature
+      )
+    },
+    createProof: (store: ProofStore, original: string) =>
+      generateERC721(store, original, Network.Mainnet),
   },
   Email: {
     ledger: env.VITE_SC_EMAIL_LEDGER_CONTRACT_ADDRESS,
     title: 'Email derivatives',
+    proofTitle: 'Goerli NFTs',
     ownerTitle: 'This wallet belongs to someone with {derivative}',
     ownerContent:
       'This is a zkNFT derivative of an email. It means this person has been verified own a ‘{original}‘ email.',
     proofIcon: EmailBadge,
     badgeType: BadgeSourceType.Email,
     network: Network.Goerli,
+    mint: createEmailBadge,
+    createProof: async (
+      store: ProofStore,
```

Comment:
I think we can serialize those args:
```TS
interface CreateProofArguments {
 store: ProofStore,
 original: string,
 options?: {secret: string}
 }
```
Use this one everywhere, maybe it's better to use object? You have 2-3 args each time

Reply:
I'm not sure if this is needed